lib/winrm/winrm_service.rb

```diff
@@ -40,20 +41,31 @@ def initialize(endpoint, transport = :kerberos, opts = {})
       @max_env_sz = DEFAULT_MAX_ENV_SIZE
       @locale = DEFAULT_LOCALE
       @logger = Logging.logger[self]
-      case transport
-      when :kerberos
-        require 'gssapi'
-        require 'gssapi/extensions'
-        @xfer = HTTP::HttpGSSAPI.new(endpoint, opts[:realm], opts[:service], opts[:keytab], opts)
-      when :plaintext
-        @xfer = HTTP::HttpPlaintext.new(endpoint, opts[:user], opts[:pass], opts)
-      when :ssl
-        @xfer = HTTP::HttpSSL.new(endpoint, opts[:user], opts[:pass], opts[:ca_trust_path], opts)
-      else
+      begin
+        @xfer = send "init_#{transport}_transport", opts.merge({endpoint: endpoint})
+      rescue Exception => e
```

Comment:
rescuing all exceptions is really dangerous.  :frowning:   Maybe this should just be `NoMethodError`?